src/dmd/utils.d

```diff
@@ -210,6 +140,79 @@ void escapePath(OutBuffer* buf, const(char)* fname)
     }
 }
 
+/**
+ * Takes a path, and make it compatible with GNU Makefile format.
+ *
+ * GNU make uses a weird quoting scheme for white space.
+ * A space or tab preceded by 2N+1 backslashes represents N backslashes followed by space;
+ * a space or tab preceded by 2N backslashes represents N backslashes at the end of a file name;
+ * and backslashes in other contexts should not be doubled.
+ *
+ * Params:
+ *   buf = Buffer to write the escaped path to
+ *   fname = Path to escape
+ */
+void escapeMakePath(OutBuffer *buf, const(char)* fname)
+{
+    uint slashes;
+
+    while(*fname)
+    {
+        switch (*fname)
+        {
+        case '\\':
+            slashes++;
+            break;
+        case '$':
+            buf.writeByte('$');
+            goto default;
+        case ' ':
+        case '\t':
+            while(slashes--)
+                buf.writeByte('\\');
+            goto case;
+        case '%':
```

Comment:
I don't understand why you are escaping `%` here.
```
test.exe: test%percent.o
	gdc $^ -o $@
test%percent.o: test%percent.d 
	gdc $^ -c -o $@
```
```
$ make -f makepercent.mak 
gdc test%percent.d -c -o test%percent.o
gdc test%percent.o -o test.exe
```

Reply:
I read it somewhere by googling about special chars escaping in GNU make, but without checking whether it is useful or not.

Reply:
I made the same tests with GNU make 4.3
Not only escaping ':' is not useful, it also breaks the Makefile.
The escaping of ':', '#' and '$' works fine.
The escaping of whitespace and non-escaping of '\' also works with proper quoting in the make commands